meta-mbl-distro/recipes-support/swupdate/swupdate_%.bbappend

```diff
@@ -2,11 +2,21 @@
 #
 # SPDX-License-Identifier: MIT
 
-# We need to configure swupdate to unset CONFIG_UBOOT and set CONFIG_BOOTLOADER_NONE
-# as we don't have a u-boot environment. Add the path to our defconfig to FILESEXTRAPATHS.
+# We need to configure swupdate to unset CONFIG_UBOOT and set CONFIG_BOOTLOADER_NONE.
+# We also remove its remote handler and webserver features.
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 
+# Tell swupdate to not try and install systemd.service files which don't exist, as we're
+# overriding do_install to remove them.
+SYSTEMD_SERVICE_${PN} = ""
 
+# Change the location of the software versions file to our user config partition.
 do_configure_prepend() {
     printf 'CONFIG_SW_VERSIONS_FILE="%s"' "${MBL_CONFIG_DIR}/sw-versions" >> "${WORKDIR}/defconfig"
 }
+
+# Override to remove the installation of all systemd.service files. We need this as swupdate's systemd
+# integration is baked in to swupdate.inc in meta-swupdate.
+do_install() {
```

Comment:
You can set `SYSTEMD_AUTO_ENABLE = "disable"`.
https://www.yoctoproject.org/docs/current/mega-manual/mega-manual.html#var-SYSTEMD_AUTO_ENABLE

Reply:
TBH, I don't think that would be better - we don't want the service at all, whether it's started by default or not.